cpp/src/plasma/eviction_policy.cc

```diff
@@ -67,32 +67,24 @@ int64_t EvictionPolicy::choose_objects_to_evict(int64_t num_bytes_required,
 void EvictionPolicy::object_created(const ObjectID& object_id) {
   auto entry = store_info_->objects[object_id].get();
   cache_.add(object_id, entry->info.data_size + entry->info.metadata_size);
+  int64_t size = entry->info.data_size + entry->info.metadata_size;
+  memory_used_ += size;
 }
 
 bool EvictionPolicy::require_space(int64_t size,
                                    std::vector<ObjectID>* objects_to_evict) {
   /* Check if there is enough space to create the object. */
   int64_t required_space = memory_used_ + size - store_info_->memory_capacity;
-  int64_t num_bytes_evicted;
-  if (required_space > 0) {
-    /* Try to free up at least as much space as we need right now but ideally
```

Comment:
Even if `required_space` is 0, we should still try to evict stuff because. This method is only being called because we need more space, which can happen even if `required_space` is 0 (due to fragmentation and things like that).